python/python/bystro/supervised_ppca/tests/test_generative_pt.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import numpy.linalg as la
+import scipy.stats as st
+from ..gf_generative_pt import PPCApt, SPCApt, FactorAnalysispt
+
+def generate_data_ppca():
+    rng = np.random.default_rng(2021)
+    N = 10000
+    L = 4
+    p = 50
+    sigma = 1.0
+    W_base = st.ortho_group.rvs(p)
+    W = W_base[:L]
+    lamb = rng.gamma(1, 1, size=L) + 1
+    for ll in range(L):
+        W[ll] = W[ll] * lamb[ll]
+    S_train = rng.normal(size=(N, L))
+    X_hat = np.dot(S_train, W)
+    X_noise = sigma * rng.normal(size=(N, p))
+    X_train = X_hat + X_noise
+    return W, sigma, X_train, S_train
+
+
+def generate_data_spca(L=4, sigmax=1.0, sigmay=0.5, N=10000):
```

Comment:
where do we use this function and the other helper function below?

Reply:
When I add the other unit tests in the next commit, that I left off for now just so everyone could see what the code looked like.

Reply:
I agree with the intention of maximizing clarity, but would recommend that all unit tests go in with the application code that they test.  IMHO it's better to have a longer PR than to separate application code and unit tests.